### PR TITLE
Compatibility with pure C11 clients

### PR DESCRIPTION
**Pull request by sleiner**
This patch makes a test file which includes all metal headers compile with the following arguments: -std=c11 -pedantic.

It is the reworked form of the draft pull request #125 

Tested with GCC 8.2.0.

### CODE REVIEW

**Comment by arnopo** [comment]
@sleiner 
For the next PR you sent, please update the on-going PR instead of closing it and opening a new one. This help to track discussions.

FYI: To update a PR by modifying existing patches you just have to update your personal github branch using " push -f" to force your branch on a new commit. Gihub will detect this and automatically update the PR and add a commentin discussions  with a "force_push" link which provides the delta.

**Comment by sleiner** [comment]
That was my original plan 🤪
I closed the other one by accident: I force-pushed the commit from where I forked off (and was planning to push the two new commits later). But that first force push closed the pull request and I could not re-open it. That's why I created this second pull request and linked it to the first one. Next time I'll know 😅

File: test/system/freertos/main.c [hunk 1]
```diff
@@ -33,7 +33,7 @@ int main(void)
 
 	/* Will not get here, unless a call is made to vTaskEndScheduler() */
 	while (1) {
-		__asm__("wfi\n\t");
```
Comment (edmooring) [inline test/system/freertos/main.c:36]: This file also needs a '#include "metal/compiler.h" to get around an implicit declaration warning.

Comment (sleiner) [reply test/system/freertos/main.c:36]: Thanks! Fixed in 2c83792487bca1e50fea2bdf7c52ccfd5608505f 👍

**Review by edmooring — APPROVED** [review]

**Pull request merged**